### PR TITLE
Fix a few tests

### PR DESCRIPTION
**Pull request by shindere**
A few tests have actually been broken by commit
b2a0f3b659a8d7e2f44c337718c1ea9373479d7d
The present PR fixes all the tests that had not been fixed meanwhile.

### CODE REVIEW

File: testsuite/tests/lib-unix/common/cloexec.ml [hunk 1]
```diff
@@ -18,7 +18,7 @@
 include unix
 files = "fdstatus_aux.c fdstatus_main.ml"
 
-**libunix
+** libunix
```
Comment (dra27) [inline testsuite/tests/lib-unix/common/cloexec.ml:21]: Is this change cosmetic, or was `**libunix` meaningless?

**Review by dra27 — APPROVED** [review]
Sorry for breaking these! However, given #7909 and #8509, I really think there needs to be an error from `ocamltest` if nothing has "happened" (I thought there was an issue tracking this, but I can't find it - perhaps I'm thinking of a previous discussion?).

Essentially, I was thinking that actions should be classified as environment or testing, so `hassysthreads` is an environment action (i.e. it doesn't perform a test) but `bytecode` and `native` are testing actions. So if you don't have systhreads compiled, then the test is *skipped*, which is OK, but otherwise the test fails unless a testing action has run and returned Success.

**Comment by shindere** [comment]
David Allsopp (2019/07/17 06:27 -0700):
> dra27 approved this pull request.

Thanks.

> Sorry for breaking these!

No problem, it's also due to the lack of documentation, which I am
currently writing.

> However, given #7909 and #8509, I really think there needs to be an
> error from `ocamltest` if nothing has "happened" (I thought there was
> an issue tracking this, but I can't find it - perhaps I'm thinking of
> a previous discussion?).
> Essentially, I was thinking that actions should be classified as
> environment or testing, so `hassysthreads` is an environment action
> (i.e. it doesn't perform a test) but `bytecode` and `native` are
> testing actions. So if you don't have systhreads compiled, then the
> test is *skipped*, which is OK, but otherwise the test fails unless a
> testing action has run and returned Success.

Yeah, it makes sense, thanks for the suggestion. Some actions would
remain difficult to classify, like the "script" action which could
belong to both categories depending on what the invoked script does, but
I guess even if it's not complete at least lcassifying the internal
actions would already improve the situation.

> > @@ -18,7 +18,7 @@
>  include unix
>  files = "fdstatus_aux.c fdstatus_main.ml"
>
> -**libunix
> +** libunix
>
> Is this change cosmetic, or was `**libunix` meaningless?

It's cosmetic, it seems the grammar does not make the space mandatory
but we have spaces everywhere.

**Comment by dra27** [comment]
> No problem, it's also due to the lack of documentation, which I am
> currently writing.

Yes, and no - as it stands, I think that even having read the manual (which of course I will have!) this is the kind of "gotcha" which would continue to happen.

> >  However, given #7909 and #8509, I really think there needs to be an
>  error from `ocamltest` if nothing has "happened" (I thought there was
>  an issue tracking this, but I can't find it - perhaps I'm thinking of
>  a previous discussion?).
>  Essentially, I was thinking that actions should be classified as
>  environment or testing, so `hassysthreads` is an environment action
>  (i.e. it doesn't perform a test) but `bytecode` and `native` are
>  testing actions. So if you don't have systhreads compiled, then the
>  test is *skipped*, which is OK, but otherwise the test fails unless a
>  testing action has run and returned Success.

> Yeah, it makes sense, thanks for the suggestion. Some actions would
> remain difficult to classify, like the "script" action which could
> belong to both categories depending on what the invoked script does, but
> I guess even if it's not complete at least lcassifying the internal
> actions would already improve the situation.

Yes, indeed - I haven't checked to see if there are tests which actually terminate with a `script` action, but perhaps here the situation could be that `script` is an environment action, but there's a "dummy" testing action `complete` which is inserted following it make it explicit that the script really is the end of the test.

>  > @@ -18,7 +18,7 @@
>   include unix
>   files = "fdstatus_aux.c fdstatus_main.ml"
> 
>  -**libunix
>  +** libunix
> 
>  Is this change cosmetic, or was `**libunix` meaningless?
> It's cosmetic, it seems the grammar does not make the space mandatory
> but we have spaces everywhere.

That's OK - I was just double-checking it wasn't a silent error!

**Comment by dra27** [comment]
4.08: 2fbab590d98adf05d992837d9daf2590d25828b3
4.09: 284332b4f193b81e0a9e2afdac64054c1dba0a84

I've pushed them directly - if it turns out those tests are failing, then I'll identify any relevant fixes (since they were already broken, we just didn't know...)

**Comment by shindere** [comment]
Thanks a lot for the merge @dra27. I have added the distinction of
actions to my ocamltest todo-list.

**Comment by shindere** [comment]
David Allsopp (2019/07/17 08:08 -0700):
> 4.08: 2fbab590d98adf05d992837d9daf2590d25828b3
> 4.09: 284332b4f193b81e0a9e2afdac64054c1dba0a84
>
> I've pushed them directly - if it turns out those tests are failing,
> then I'll identify any relevant fixes (since they were already broken,
> we just didn't know...)

Thanks a lot for your follow-up on this!

**Comment by dra27** [comment]
No problem, @shindere - 4.08 and 4.09 are still building fine. There were 3 failures on `trunk` - 2 I hope were unrelated and transient (the cygwin64 and arm64 failures), but the mingw 32bit failure in `testsuite/tests/lib-threads/beat.ml` will be because that test has been quiet for a few months because it wasn't working!

**Pull request merged**